### PR TITLE
Increase code coverage for Platform project

### PR DESCRIPTION
**Pull request by WolfyUK**
### Context
[AB#222532](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/222532) [AB#222495](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/222495)

### Change proposed in this pull request
Increase Platform unit test coverage and re-enable quality check in pipeline. Achieved via exclusion of .NET 8 generated code in `.runsettings` file, `[ExcludeFromCodeCoverage]` attributes and implementation of missing unit tests.

image

### Guidance to review 
Run unit tests with code coverage and/or trigger a new pipeline.

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: platform/Platform.sln.DotSettings [hunk 1]
```diff
@@ -1,2 +1,5 @@
 ﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
-	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=FTE/@EntryIndexedValue">FTE</s:String></wpf:ResourceDictionary>
\ No newline at end of file
+	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=FTE/@EntryIndexedValue">FTE</s:String>
+	<s:Boolean x:Key="/Default/Environment/Filtering/FileMasksToExclude/=_002A_002Eg_002Ecs/@EntryIndexedValue">True</s:Boolean>
+	<s:String x:Key="/Default/Housekeeping/UnitTestingMru/UnitTestRunner/RunConfigurationFilename/@EntryValue">C:\Users\mwoolfson\Documents\code\education-benchmarking-and-insights\platform\tests\coverlet.runsettings</s:String>
```
Comment (jrabbott) [inline platform/Platform.sln.DotSettings:4]: Need to have a look at this file reference.

Comment (WolfyUK) [reply platform/Platform.sln.DotSettings:4]: Hmm. I think this is redundant now, as Rider will use `dotCover` over `dotnet test --collect`. I'll remove

**Review by jrabbott — COMMENTED** [review]
Just the file path in the dotsettings otherwise LGTM

File: platform/Platform.sln.DotSettings (hunk 1 above)
Comment (WolfyUK) [reply platform/Platform.sln.DotSettings]: Removed

**Review by jrabbott — APPROVED** [review]

**Pull request merged**